asserts/account_key.go

```diff
@@ -48,12 +48,22 @@ func (ak *AccountKey) Until() time.Time {
 	return ak.until
 }
 
+// Fingerprint returns the fingerprint of the account key.
+func (ak *AccountKey) Fingerprint() string {
+	return ak.pubKey.Fingerprint()
+}
+
 // IsValidAt returns whether the account key is valid at 'when' time.
 func (ak *AccountKey) IsValidAt(when time.Time) bool {
 	return (when.After(ak.since) || when.Equal(ak.since)) && when.Before(ak.until)
 }
 
-func checkPublicKey(ab *AssertionBase, fingerprintName string) (PublicKey, error) {
+// Verify verifies signature is valid for content using the account key.
+func (ak *AccountKey) Verify(content []byte, sig Signature) error {
```

Comment:
@mvo5 yes, notice that actually AccountKey has a fingerprint field anyway so Fingerprint should be there in any case. Maybe later we really want AccountKey not to be a PublicKey but I'm still unconvinced there is conceptual clarity in that, it would need more code complexity for sure.